front-end/src/app/shared/services/committee-account.service.ts

```diff
@@ -1,20 +1,27 @@
 import { Injectable } from '@angular/core';
-import { Observable } from 'rxjs';
-import { ApiService } from './api.service';
+import { concatMap, Observable } from 'rxjs';
 import { CommitteeAccount } from '../models/committee-account.model';
+import { FecApiService } from './FecApiService/fec-api.service';
+import { Store } from '@ngrx/store';
+import { selectUserLoginData } from 'app/store/login.selectors';
 
 @Injectable({
   providedIn: 'root',
 })
 export class CommitteeAccountService {
-  constructor(private apiService: ApiService) {}
+  constructor(private fecApiService: FecApiService, private store: Store) {}
 
   /**
    * Gets the commitee account details.
    *
    * @return     {Observable}  The commitee details.
    */
   public getDetails(): Observable<CommitteeAccount> {
-    return this.apiService.get<CommitteeAccount>('/core/get_committee_details');
+      const userLoginData$ = this.store.select(selectUserLoginData);
+      return userLoginData$.pipe(
+        concatMap(userLoginData => {
+          return this.fecApiService.getDetails(userLoginData.committee_id);
+        })
+      );
```

Comment:
I had to look at this and how it is pulled into the ngrx effect a couple of times to sort out the mergeMap and concatMap are working together

Reply:
is mergemap a better option?